src/watts/plugin_sas.py

```diff
@@ -0,0 +1,203 @@
+
+        self.sas_inp_name = "SAS.inp"
+        self.show_stdout = show_stdout
+        self.show_stderr = show_stderr
+
+    @property
+    def sas_exec(self) -> Path:
+        return self._sas_exec
+
+    @property
+    def conv_channel(self) -> Path:
+        return self._conv_channel
+
+    @property
+    def conv_primar4(self) -> Path:
+        return self._conv_primar4
+
+    @sas_exec.setter
+    def sas_exec(self, exe: PathLike):
+        if shutil.which(exe) is None:
+            raise RuntimeError(f"SAS executable '{exe}' is missing.")
+        self._sas_exec = Path(exe)
+
+    @conv_channel.setter
+    def conv_channel(self, exe: PathLike):
+        if shutil.which(exe) is None:
+            raise RuntimeError(f"CHANNELtoCSV utility executable '{exe}' is missing.")
+        self._conv_channel = Path(exe)
+
+    @conv_primar4.setter
+    def conv_primar4(self, exe: PathLike):
+        if shutil.which(exe) is None:
+            raise RuntimeError(f"PRIMAR4toCSV utility executable '{exe}' is missing.")
+        self._conv_primar4 = Path(exe)
+
+    def prerun(self, params: Parameters):
+        """Generate the SAS input based on the template
+
+        Parameters
+        ----------
+        params
+            Parameters used when rendering template
+        """
+        # Render the template
+        # Make a copy of params and convert units if necessary
+        # The original params remains unchanged
+
+        params_copy = params.convert_units()
+
+        print("Pre-run for SAS Plugin")
+        self._run_time = time.time_ns()
+        super().prerun(params_copy, filename=self.sas_inp_name)
+
+    def run(self):
+        """Run SAS"""
+        print("Run for SAS Plugin")
+
+        log_file = Path("SAS_log.txt")
+
+        with log_file.open("w") as outfile:
+            func_stdout = tee_stdout if self.show_stdout else redirect_stdout
+            func_stderr = tee_stderr if self.show_stderr else redirect_stderr
+            with func_stdout(outfile), func_stderr(outfile):
+                run_proc([self.sas_exec, "-i", self.sas_inp_name, "-o", "out.txt"])
+
+    def postrun(self, params: Parameters) -> ResultsSAS:
+        """Read SAS results and create results object
+
+        Parameters
+        ----------
+        params
+            Parameters used to create SAS model
+
+        Returns
+        -------
+        SAS results object
+        """
+        print("Post-run for SAS Plugin")
+
+        # Convert CHANNEl.dat and PRIMER4.dat to csv files
+        # using SAS utilities. Check if files exist because
+        # they may not be outputted per user's choice.
+        # Doesn't work with run_proc(). Use os.system() for now.
+        # Needs to convert PosixPath back to string.
+        if Path("CHANNEL.dat").is_file():
+            os.system(str(self.conv_channel) + " <CHANNEL.dat> CHANNEL.csv") 
+        if Path("PRIMAR4.dat").is_file():
+            os.system(str(self.conv_primar4) + " <PRIMAR4.dat> PRIMAR4.csv") 
```

Comment:
You should use the [`subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run) function instead of `os.system`, which is not portable.